--- org/gudy/azureus2/ui/swt/mainwindow/MainMenu.java.orig	2007-01-12 17:51:47.000000000 -0800
+++ org/gudy/azureus2/ui/swt/mainwindow/MainMenu.java	2007-03-29 12:55:44.000000000 -0700
@@ -200,21 +200,10 @@
       addCloseDetailsMenuItem(fileMenu);
       addCloseDownloadBarsToMenu(fileMenu);
 
-
       //No need for restart and exit on OS X
       if(!Constants.isOSX) {
         new MenuItem(fileMenu, SWT.SEPARATOR);
 
-        MenuItem file_restart = new MenuItem(fileMenu, SWT.NULL);
-        Messages.setLanguageText(file_restart, "MainWindow.menu.file.restart"); //$NON-NLS-1$
-
-        file_restart.addListener(SWT.Selection, new Listener() {
-
-        public void handleEvent(Event event) {
-						UIFunctionsManagerSWT.getUIFunctionsSWT().dispose(true, false);
-					}
-				});
-
         final MenuItem file_exit = new MenuItem(fileMenu, SWT.NULL);
         if(!COConfigurationManager.getBooleanParameter("Enable System Tray") || !COConfigurationManager.getBooleanParameter("Close To Tray")) {
             KeyBindings.setAccelerator(file_exit, "MainWindow.menu.file.exit");
@@ -381,26 +370,6 @@
 			menu_plugin_logViews.setMenu(pluginLogsMenu);
 			menu_plugin_logViews.setData("EOL", "1");
 
-      new MenuItem(pluginMenu, SWT.SEPARATOR);
-      
-      MenuItem plugins_install_wizard = new MenuItem(pluginMenu, SWT.NULL);
-      KeyBindings.setAccelerator(plugins_install_wizard, "MainWindow.menu.plugins.installPlugins");
-      Messages.setLanguageText(plugins_install_wizard, "MainWindow.menu.plugins.installPlugins"); //$NON-NLS-1$
-      plugins_install_wizard.addListener(SWT.Selection, new Listener() {
-        public void handleEvent(Event e) {
-          new InstallPluginWizard(core, display);
-        }
-      });
-      
-      MenuItem plugins_uninstall_wizard = new MenuItem(pluginMenu, SWT.NULL);
-      KeyBindings.setAccelerator(plugins_uninstall_wizard, "MainWindow.menu.plugins.uninstallPlugins");
-      Messages.setLanguageText(plugins_uninstall_wizard, "MainWindow.menu.plugins.uninstallPlugins"); //$NON-NLS-1$
-      plugins_uninstall_wizard.addListener(SWT.Selection, new Listener() {
-        public void handleEvent(Event e) {
-          new UnInstallPluginWizard(core, display);
-        }
-      });
-      
       // standard items
       if(Constants.isOSX) {
           // Window menu
@@ -478,49 +447,8 @@
         }
       });
 
-
-      MenuItem help_plugin= new MenuItem(helpMenu, SWT.NULL);
-      Messages.setLanguageText(help_plugin, "MainWindow.menu.help.plugins"); //$NON-NLS-1$
-      help_plugin.addListener(SWT.Selection, new Listener() {
-          public void handleEvent(Event e) {
-            String pluginString = "http://azureus.sourceforge.net/plugin_list.php";
-            Utils.launch(pluginString);
-          }
-        });
-      
       new MenuItem(helpMenu,SWT.SEPARATOR);
       
-      if ( !SystemProperties.isJavaWebStartInstance()){
-        MenuItem help_checkupdate = new MenuItem(helpMenu, SWT.NULL);
-        KeyBindings.setAccelerator(help_checkupdate, "MainWindow.menu.help.checkupdate");
-        Messages.setLanguageText(help_checkupdate, "MainWindow.menu.help.checkupdate"); //$NON-NLS-1$
-        help_checkupdate.addListener(SWT.Selection, new Listener() {
-        	public void handleEvent(Event e) {
-        		UIFunctionsSWT uiFunctions = UIFunctionsManagerSWT.getUIFunctionsSWT();
-        		if (uiFunctions != null) {
-        			uiFunctions.bringToFront();
-        		}
-        		UpdateMonitor.getSingleton(core).performCheck(true, false, new UpdateCheckInstanceListener() {
-        			public void cancelled(UpdateCheckInstance instance) {
-        			}
-        			
-        			public void complete(UpdateCheckInstance instance) {
-        				if (instance.getUpdates().length == 0) {
-        					Utils.execSWTThread(new AERunnable() {
-        						public void runSupport() {
-            					Utils.openMessageBox(parent,
-													SWT.ICON_INFORMATION | SWT.OK,
-													"window.update.noupdates", (String[]) null);
-        						}
-        					});
-        				}
-        			}
-        		});
-        	}
-        });
-      }
-
-      new MenuItem(helpMenu,SWT.SEPARATOR);
       MenuItem help_debug = new MenuItem(helpMenu, SWT.NULL);
       Messages.setLanguageText(help_debug, "MainWindow.menu.help.debug");
       help_debug.addListener(SWT.Selection, new Listener() {
