--- org/gudy/azureus2/ui/swt/Messages.java.orig	2006-01-31 10:12:35.000000000 -0500
+++ org/gudy/azureus2/ui/swt/Messages.java	2006-02-07 16:14:10.000000000 -0500
@@ -203,23 +203,6 @@
           ((ToolItem)widget).setToolTipText(toolTip);
         }
       }
-    } else if (widget instanceof TableColumn) {
-      String key = (String) widget.getData();
-			if (key != null) {
-				if (!key.endsWith(".info"))
-					key += ".info";
-				String toolTip = MessageText.getString(key, (String) null);
-				if (toolTip == null)
-					toolTip = MessageText.getString(key.substring(0, key.length() - 5),
-							(String) null);
-				if (toolTip != null) {
-					try {
-						((TableColumn) widget).setToolTipText(toolTip);
-					} catch (NoSuchMethodError e) {
-						// Pre SWT 3.2
-					}
-				}
-			}
     }
   }
   
