--- org/gudy/azureus2/core3/config/impl/ConfigurationManager.java~	2005-10-03 09:36:58.000000000 -0700
+++ org/gudy/azureus2/core3/config/impl/ConfigurationManager.java	2005-12-18 07:30:07.000000000 -0800
@@ -201,14 +201,14 @@
   }
   
   private String getStringParameter(String parameter, byte[] defaultValue) {
-    try {
-      return new String(getByteParameter(parameter, defaultValue));
-    } catch (Exception e) {
-      byte[] bytesReturn = getByteParameter(parameter, null);
-      if (bytesReturn == null)
-        return null;
-      return new String(bytesReturn);
-    }
+    byte[] bp = getByteParameter(parameter, defaultValue);
+    if (bp == null)
+      {
+	bp = getByteParameter(parameter, null);
+	if (bp == null)
+	  return null;
+      }
+    return new String(bp);
   }
   
   public String getStringParameter(String parameter, String defaultValue) {
