diff -up azureus-4.0.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java.no-updates-PluginInitializer azureus-4.0.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java
--- azureus-4.0.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java.no-updates-PluginInitializer	2008-10-31 16:25:38.000000000 +0200
+++ azureus-4.0.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java	2009-09-02 21:53:06.000000000 +0300
@@ -44,13 +44,8 @@ import org.gudy.azureus2.core3.logging.*
 import org.gudy.azureus2.plugins.*;
 import org.gudy.azureus2.pluginsimpl.local.launch.PluginLauncherImpl;
 import org.gudy.azureus2.pluginsimpl.local.ui.UIManagerImpl;
-import org.gudy.azureus2.pluginsimpl.local.update.*;
 import org.gudy.azureus2.pluginsimpl.local.utils.UtilitiesImpl;
 
-import org.gudy.azureus2.update.UpdaterUpdateChecker;
-import org.gudy.azureus2.update.UpdaterUtils;
-
-
 
 /**
  * @author Olivier
@@ -95,12 +90,12 @@ PluginInitializer
    					"ShareHoster",
 					"true",
 					"false"},
-   			{	 PluginManagerDefaults.PID_PLUGIN_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.pluginsimpl.update.PluginUpdatePlugin", 
-   					"azbppluginupdate", 
-   					"PluginUpdate",
-					"true",
-					"true"},
+//    			{	 PluginManagerDefaults.PID_PLUGIN_UPDATE_CHECKER, 
+//    					"org.gudy.azureus2.pluginsimpl.update.PluginUpdatePlugin", 
+//    					"azbppluginupdate", 
+//    					"PluginUpdate",
+// 					"true",
+// 					"true"},
 			{	 PluginManagerDefaults.PID_UPNP, 
 				    "com.aelitis.azureus.plugins.upnp.UPnPPlugin", 
 				    "azbpupnp", 
@@ -125,18 +120,18 @@ PluginInitializer
 					"Magnet URI Handler",
 					"true",
 					"false"},
-			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.update.CoreUpdateChecker", 
-   					"azbpcoreupdater", 
-   					"CoreUpdater",
-					"true",
-					"true"},
-			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
-   					"org.gudy.azureus2.update.CorePatchChecker", 
-   					"azbpcorepatcher", 
-   					"CorePatcher",
-					"true",
-					"true"},
+// 			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
+//    					"org.gudy.azureus2.update.CoreUpdateChecker", 
+//    					"azbpcoreupdater", 
+//    					"CoreUpdater",
+// 					"true",
+// 					"true"},
+// 			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
+//    					"org.gudy.azureus2.update.CorePatchChecker", 
+//    					"azbpcorepatcher", 
+//    					"CorePatcher",
+// 					"true",
+// 					"true"},
 	   		{	 PluginManagerDefaults.PID_PLATFORM_CHECKER, 
    					"org.gudy.azureus2.platform.PlatformManagerPluginDelegate", 
    					"azplatform2", 
@@ -420,11 +415,11 @@ PluginInitializer
   	
   	core_operation 	= _core_operation;
     
-    UpdateManagerImpl.getSingleton( azureus_core );	// initialise the update manager
+	//    UpdateManagerImpl.getSingleton( azureus_core );	// initialise the update manager
        
     plugin_manager = PluginManagerImpl.getSingleton( this );
     
-    UpdaterUtils.checkBootstrapPlugins();
+    //   UpdaterUtils.checkBootstrapPlugins();
   }
   
   protected void
@@ -1122,8 +1117,8 @@ PluginInitializer
 	      if ( load_failure != null ){
 	    	  plugin_interface.setAsFailed();
 	    	  
-	    	// don't complain about our internal one
-	    	  if ( !pid.equals(UpdaterUpdateChecker.getPluginID())){
+		  //	    	// don't complain about our internal one
+		  //	    	  if ( !pid.equals(UpdaterUpdateChecker.getPluginID())){
 	    		  
 		      	String msg = "Error loading plugin '" + pluginName + "' / '" + plugin_class_string + "'";
 		      	LogAlert la;
@@ -1136,7 +1131,7 @@ PluginInitializer
 		      	Logger.log(la);
 	
 		      	System.out.println( msg + ": " + load_failure);
-		      }
+			//		      }
 	      }
   		}
 	      
