diff -up azureus-4.6.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java.no-updates-PluginInitializer azureus-4.6.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java
--- azureus-4.6.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java.no-updates-PluginInitializer	2011-03-08 19:09:54.000000000 +0200
+++ azureus-4.6.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java	2011-04-02 17:15:52.000000000 +0300
@@ -41,12 +41,8 @@ import org.gudy.azureus2.core3.util.*;
 import org.gudy.azureus2.plugins.*;
 import org.gudy.azureus2.pluginsimpl.local.launch.PluginLauncherImpl;
 import org.gudy.azureus2.pluginsimpl.local.ui.UIManagerImpl;
-import org.gudy.azureus2.pluginsimpl.local.update.UpdateManagerImpl;
 import org.gudy.azureus2.pluginsimpl.local.utils.UtilitiesImpl;
 import org.gudy.azureus2.pluginsimpl.local.utils.UtilitiesImpl.runnableWithException;
-import org.gudy.azureus2.update.UpdaterUpdateChecker;
-import org.gudy.azureus2.update.UpdaterUtils;
-
 
 import com.aelitis.azureus.core.*;
 import com.aelitis.azureus.core.versioncheck.VersionCheckClient;
@@ -98,12 +94,12 @@ PluginInitializer
    					"ShareHoster",
 					"true",
 					"false"},
-   			{	 PluginManagerDefaults.PID_PLUGIN_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.pluginsimpl.update.PluginUpdatePlugin", 
-   					"azbppluginupdate", 
-   					"PluginUpdate",
-					"true",
-					"true"},
+//    			{	 PluginManagerDefaults.PID_PLUGIN_UPDATE_CHECKER, 
+//    					"org.gudy.azureus2.pluginsimpl.update.PluginUpdatePlugin", 
+//    					"azbppluginupdate", 
+//    					"PluginUpdate",
+// 					"true",
+// 					"true"},
 			{	 PluginManagerDefaults.PID_UPNP, 
 				    "com.aelitis.azureus.plugins.upnp.UPnPPlugin", 
 				    "azbpupnp", 
@@ -128,18 +124,18 @@ PluginInitializer
 					"Magnet URI Handler",
 					"true",
 					"false"},
-			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.update.CoreUpdateChecker", 
-   					"azbpcoreupdater", 
-   					"CoreUpdater",
-					"true",
-					"true"},
-			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
-   					"org.gudy.azureus2.update.CorePatchChecker", 
-   					"azbpcorepatcher", 
-   					"CorePatcher",
-					"true",
-					"true"},
+// 			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
+//    					"org.gudy.azureus2.update.CoreUpdateChecker", 
+//    					"azbpcoreupdater", 
+//    					"CoreUpdater",
+// 					"true",
+// 					"true"},
+// 			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
+//    					"org.gudy.azureus2.update.CorePatchChecker", 
+//    					"azbpcorepatcher", 
+//    					"CorePatcher",
+// 					"true",
+// 					"true"},
 	   		{	 PluginManagerDefaults.PID_PLATFORM_CHECKER, 
    					"org.gudy.azureus2.platform.PlatformManagerPluginDelegate", 
    					"azplatform2", 
@@ -458,7 +454,7 @@ PluginInitializer
   	
   	core_operation 	= _core_operation;
     
-    UpdateManagerImpl.getSingleton( azureus_core );	// initialise the update manager
+//     UpdateManagerImpl.getSingleton( azureus_core );	// initialise the update manager
        
     plugin_manager = PluginManagerImpl.getSingleton( this );
     
@@ -480,7 +476,7 @@ PluginInitializer
     	}
     }
     
-    UpdaterUtils.checkBootstrapPlugins();
+//     UpdaterUtils.checkBootstrapPlugins();
   }
   
   protected void
@@ -1291,9 +1287,6 @@ PluginInitializer
 					  if ( load_failure != null ){
 						  plugin_interface.setAsFailed();
 
-						  // don't complain about our internal one
-						  if ( !pid.equals(UpdaterUpdateChecker.getPluginID())){
-
 							  String msg = "Error loading plugin '" + pluginName + "' / '" + plugin_class_string + "'";
 							  LogAlert la;
 							  if (load_failure instanceof UnsupportedClassVersionError) {
@@ -1305,7 +1298,6 @@ PluginInitializer
 							  Logger.log(la);
 
 							  System.out.println( msg + ": " + load_failure);
-						  }
 					  }
 				  }
 			  }
