--- org/gudy/azureus2/pluginsimpl/local/installer/PluginInstallerImpl.java.orig	2006-02-09 23:42:33.000000000 -0500
+++ org/gudy/azureus2/pluginsimpl/local/installer/PluginInstallerImpl.java	2006-02-09 23:42:45.000000000 -0500
@@ -237,13 +237,6 @@
 					PluginManager.registerPlugin( dummy_plugin, plugin_id );
 				
 					PluginInterface dummy_plugin_interface = manager.getPluginInterfaceByID( plugin_id );
-					
-					((InstallablePluginImpl)plugin).addUpdate( inst, pup, dummy_plugin, dummy_plugin_interface );
-							
-				}else{
-					
-					((InstallablePluginImpl)plugin).addUpdate( inst, pup, existing_plugin, existing_plugin_interface );
-	
 				}
 			}
 		
