--- org/gudy/azureus2/ui/swt/views/configsections/ConfigSectionPlugins.java.~1.7.~	2005-12-21 07:34:08.000000000 -0800
+++ org/gudy/azureus2/ui/swt/views/configsections/ConfigSectionPlugins.java	2006-01-25 14:40:42.000000000 -0800
@@ -373,12 +373,6 @@
 			tc.addSelectionListener(new SelectionAdapter() {
 				public void widgetSelected(SelectionEvent e) {
 					boolean ascending = comparator.setField(table.indexOf(tc));
-					try {
-						table.setSortColumn(tc);
-						table.setSortDirection(ascending ? SWT.UP : SWT.DOWN);
-					} catch (NoSuchMethodError ignore) {
-						// Ignore Pre 3.0
-					}
 					Collections.sort(pluginIFs, comparator);
 					table.clearAll();
 				}
