--- azureus-5.6.orig/org/gudy/azureus2/pluginsimpl/local/utils/xml/simpleparser/SimpleXMLParserDocumentImpl.java	2015-05-05 02:29:02.569602779 +0100
+++ azureus-5.6/org/gudy/azureus2/pluginsimpl/local/utils/xml/simpleparser/SimpleXMLParserDocumentImpl.java	2015-05-05 02:29:33.439030178 +0100
@@ -25,7 +25,6 @@ package org.gudy.azureus2.pluginsimpl.lo
 import javax.xml.parsers.*;
 
 import org.xml.sax.*;
-import org.apache.commons.lang.Entities;
 import org.gudy.azureus2.core3.util.AENetworkClassifier;
 import org.gudy.azureus2.core3.util.Constants;
 import org.gudy.azureus2.core3.util.Debug;
@@ -646,28 +645,7 @@ SimpleXMLParserDocumentImpl
 		     				
 		     				String	replacement;
 		     				
-		     				if ( 	ref.equals( "amp") 		|| 
-		     						ref.equals( "lt" ) 		|| 
-		     						ref.equals( "gt" ) 		|| 
-		     						ref.equals( "quot" )	|| 
-		     						ref.equals( "apos" ) 	||
-		     						ref.startsWith( "#" )){
-		     					
-		     					replacement = new String( buffer, 0, buffer_pos );
-		     					
-		     				}else{
-		     							     							     					
-			     				int num = Entities.HTML40.entityValue( ref );
-			     					
-		     					if ( num != -1 ){
-		     					
-		     						replacement = "&#" + num + ";";
-		     						
-		     					}else{
-		     						
-		     						replacement = new String( buffer, 0, buffer_pos );
-		     					}
-		     				}
+                            replacement = new String( buffer, 0, buffer_pos );
 		     				
 		     				char[] chars = replacement.toCharArray();
 		     				
