--- org/gudy/azureus2/ui/console/multiuser/UserManager.java~	2005-09-06 08:44:54.000000000 -0700
+++ org/gudy/azureus2/ui/console/multiuser/UserManager.java	2005-12-17 12:06:13.000000000 -0800
@@ -23,7 +23,7 @@
 package org.gudy.azureus2.ui.console.multiuser;
 
 import java.beans.XMLDecoder;
-import java.beans.XMLEncoder;
+// import java.beans.XMLEncoder;
 import java.io.BufferedInputStream;
 import java.io.BufferedOutputStream;
 import java.io.File;
@@ -159,9 +159,9 @@
 		List users = new ArrayList( usersMap.values() );
 		config.setUsers(users);
 		
-		XMLEncoder encoder = new XMLEncoder( new BufferedOutputStream( out ) );
-		encoder.writeObject(config);
-		encoder.close();
+		// XMLEncoder encoder = new XMLEncoder( new BufferedOutputStream( out ) );
+		// encoder.writeObject(config);
+		// encoder.close();
 	}
 
 	public static UserManager getInstance(PluginInterface pi) 
