--- ../azureus3orig/azureus3/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java	2007-12-03 10:07:11.000000000 -0500
+++ org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java	2007-12-03 10:27:39.000000000 -0500
@@ -44,13 +44,8 @@
 import org.gudy.azureus2.plugins.*;
 import org.gudy.azureus2.pluginsimpl.local.launch.PluginLauncherImpl;
 import org.gudy.azureus2.pluginsimpl.local.ui.UIManagerImpl;
-import org.gudy.azureus2.pluginsimpl.local.update.*;
 import org.gudy.azureus2.pluginsimpl.local.utils.UtilitiesImpl;
 
-import org.gudy.azureus2.update.UpdaterUpdateChecker;
-import org.gudy.azureus2.update.UpdaterUtils;
-
-
 
 /**
  * @author Olivier
@@ -91,11 +86,6 @@
    					"azbpsharehoster", 
    					"ShareHoster",
 					"true" },
-   			{	 PluginManagerDefaults.PID_PLUGIN_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.pluginsimpl.update.PluginUpdatePlugin", 
-   					"azbppluginupdate", 
-   					"PluginUpdate",
-					"true" },
 			{	 PluginManagerDefaults.PID_CLIENT_ID, 
 				    "com.aelitis.azureus.plugins.clientid.ClientIDPlugin", 
 				    "azbpclientid", 
@@ -121,16 +111,6 @@
 					"azbpmagnet", 
 					"Magnet URI Handler",
 					"true" },
-			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.update.CoreUpdateChecker", 
-   					"azbpcoreupdater", 
-   					"CoreUpdater",
-					"true" },
-			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
-   					"org.gudy.azureus2.update.CorePatchChecker", 
-   					"azbpcorepatcher", 
-   					"CorePatcher",
-					"true" },
 	   		{	 PluginManagerDefaults.PID_PLATFORM_CHECKER, 
    					"org.gudy.azureus2.platform.PlatformManagerPluginDelegate", 
    					"azplatform2", 
@@ -316,11 +296,7 @@
   	
   	core_operation 	= _core_operation;
     
-    UpdateManagerImpl.getSingleton( azureus_core );	// initialise the update manager
-       
     plugin_manager = PluginManagerImpl.getSingleton( this );
-    
-    UpdaterUtils.checkBootstrapPlugins();
   }
   
   protected boolean
@@ -946,16 +922,11 @@
 	      
 	      if ( load_failure != null ){
 	      		  
-	    	  	// don't complain about our internal one
-	    	  
-	    	  if ( !pid.equals(UpdaterUpdateChecker.getPluginID())){
-	    		  
 		      	String	msg = "Error loading plugin '" + pluginName + "' / '" + plugin_class_string + "'";
 		   	 
 		      	Logger.log(new LogAlert(LogAlert.UNREPEATABLE, msg, load_failure));
 	
 		      	System.out.println( msg + " : " + load_failure);
-		      }
 	      }
   		}
 	      
