--- com/aelitis/azureus/core/impl/AzureusCoreImpl.java.orig	2006-03-25 14:17:49.000000000 -0800
+++ com/aelitis/azureus/core/impl/AzureusCoreImpl.java	2006-03-25 15:14:01.000000000 -0800
@@ -584,14 +584,6 @@
 			if (Logger.isEnabled())
 				Logger.log(new LogEvent(LOGID, "Stop operation completes"));
 	
-				// if any installers exist then we need to closedown via the updater
-			
-			if ( 	apply_updates && 
-					getPluginManager().getDefaultPluginInterface().getUpdateManager().getInstallers().length > 0 ){
-				
-				AzureusRestarterFactory.create( this ).restart( true );
-			}
-			
 			try{
 				ThreadGroup	tg = Thread.currentThread().getThreadGroup();
 				
