--- org/gudy/azureus2/ui/swt/mainwindow/MainStatusBar.java.~1.1.~	2006-03-20 23:21:17.000000000 -0800
+++ org/gudy/azureus2/ui/swt/mainwindow/MainStatusBar.java	2006-03-25 18:13:47.000000000 -0800
@@ -544,8 +544,6 @@
 		});
 		statusDown.setMenu(menuDownSpeed);
 
-		addUpdateListener();
-
 		return statusBar;
 	}
 
@@ -619,16 +617,6 @@
 		}
 	}
 
-	private void addUpdateListener() {
-		azureusCore.getPluginManager().getDefaultPluginInterface()
-				.getUpdateManager().addListener(new UpdateManagerListener() {
-					public void checkInstanceCreated(UpdateCheckInstance instance) {
-
-						new updateStatusChanger(instance);
-					}
-				});
-	}
-
 	protected class updateStatusChanger {
 		UpdateCheckInstance instance;
 
