--- com/aelitis/azureus/core/diskmanager/MemoryMappedFile.java~	2004-09-23 08:52:12.000000000 -0700
+++ com/aelitis/azureus/core/diskmanager/MemoryMappedFile.java	2005-12-17 12:03:21.000000000 -0800
@@ -220,8 +220,8 @@
   				try {
   					Method getCleanerMethod = buffer.getClass().getMethod( "cleaner", new Class[0] );
   					getCleanerMethod.setAccessible( true );
-  					sun.misc.Cleaner cleaner = (sun.misc.Cleaner)getCleanerMethod.invoke( buffer, new Object[0] );
-  					cleaner.clean();
+					// sun.misc.Cleaner cleaner = (sun.misc.Cleaner)getCleanerMethod.invoke( buffer, new Object[0] );
+  					// cleaner.clean();
   				}
           catch (Exception e) { Debug.printStackTrace( e ); }
   				return null;
