--- org/gudy/azureus2/core3/security/impl/SESecurityManagerImpl.java~	2006-01-17 21:58:55.000000000 -0800
+++ org/gudy/azureus2/core3/security/impl/SESecurityManagerImpl.java	2006-01-17 22:20:05.000000000 -0800
@@ -94,7 +94,7 @@
 		}
 		
 		try{
-			SESecurityManagerBC.initialise();
+		    //			SESecurityManagerBC.initialise();
 			
 		}catch( Throwable e ){
 			
@@ -505,7 +505,8 @@
 	
 		throws Exception
 	{
-		return( SESecurityManagerBC.createSelfSignedCertificate( this, alias, cert_dn, strength ));
+	    //		return( SESecurityManagerBC.createSelfSignedCertificate( this, alias, cert_dn, strength ));
+	    return null;
 	}
 	
 	public boolean
