--- org/gudy/azureus2/core3/config/COConfigurationManager.java.orig	2007-11-21 22:46:58.000000000 -0500
+++ org/gudy/azureus2/core3/config/COConfigurationManager.java	2008-07-14 15:48:16.000000000 -0400
@@ -40,7 +40,7 @@ COConfigurationManager 
 	public static final int CONFIG_CACHE_SIZE_MAX_MB;
 	
 	static{
-		long max_mem_bytes 	= Runtime.getRuntime().maxMemory();
+	    long max_mem_bytes 	= Math.min (Runtime.getRuntime().maxMemory(), 64*1024*1024);
 	    long mb_1			= 1*1024*1024;
 	    long mb_32			= 32*mb_1;
 	    int size = (int)(( max_mem_bytes - mb_32 )/mb_1);	    
