diff -up azureus-4.0.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInterfaceImpl.java.no-updates-PluginInterfaceImpl azureus-4.0.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInterfaceImpl.java
--- azureus-4.0.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInterfaceImpl.java.no-updates-PluginInterfaceImpl	2008-08-20 08:20:26.000000000 +0300
+++ azureus-4.0.0.4/org/gudy/azureus2/pluginsimpl/local/PluginInterfaceImpl.java	2009-09-03 20:35:39.000000000 +0300
@@ -50,7 +50,6 @@ import org.gudy.azureus2.pluginsimpl.loc
 import org.gudy.azureus2.pluginsimpl.local.ui.config.ParameterRepository;
 import org.gudy.azureus2.pluginsimpl.local.ui.config.PluginConfigUIFactoryImpl;
 import org.gudy.azureus2.pluginsimpl.local.utils.*;
-import org.gudy.azureus2.pluginsimpl.local.update.*;
 import org.gudy.azureus2.plugins.ipc.IPCInterface;
 import org.gudy.azureus2.plugins.ipfilter.IPFilter;
 import org.gudy.azureus2.plugins.tracker.Tracker;
@@ -426,7 +425,7 @@ PluginInterfaceImpl 
   public UpdateManager
   getUpdateManager()
   {
-  	return( UpdateManagerImpl.getSingleton( initialiser.getAzureusCore()));
+  	return( null );
   }
 
   
