--- Vuze_5750_source/com/aelitis/azureus/util/ImportExportUtils.java.orig	2016-11-29 04:51:14.000000000 +0000
+++ Vuze_5750_source/com/aelitis/azureus/util/ImportExportUtils.java	2017-03-13 00:56:15.678365502 +0000
@@ -368,7 +368,7 @@ public final class ImportExportUtils {
 		String		key,
 		String[]	data )
 	{
-		List	l = new JSONArray(data.length);
+		List	l = new JSONArray();
 		
 		map.put( key, l );
 
--- Vuze_5750_source/com/aelitis/azureus/util/JSONUtils.java.orig	2016-11-15 19:01:18.000000000 +0000
+++ Vuze_5750_source/com/aelitis/azureus/util/JSONUtils.java	2017-03-13 00:56:15.678365502 +0000
@@ -71,7 +71,7 @@ public class JSONUtils
 	 * @since 3.0.1.5
 	 */
 	public static JSONObject encodeToJSONObject(Map map) {
-		JSONObject newMap = new JSONObject((int)(map.size()*1.5));
+		JSONObject newMap = new JSONObject();
 
 		for (Map.Entry<String, Object> entry: ((Map<String,Object>)map).entrySet()){
 			String key 		= entry.getKey();
@@ -102,9 +102,7 @@ public class JSONUtils
 	 */
 	public static String encodeToJSON(Map map) {
 		JSONObject jobj = encodeToJSONObject(map);		
-		StringBuilder	sb = new StringBuilder(8192);
-		jobj.toString( sb );
-		return( sb.toString());
+		return( jobj.toString());
 	}
 
 	public static String encodeToJSON(Collection list) {
@@ -156,7 +154,7 @@ public class JSONUtils
 	 * @since 3.0.1.5
 	 */
 	private static JSONArray encodeToJSONArray(Collection list) {
-		JSONArray newList = new JSONArray(list.size());
+		JSONArray newList = new JSONArray();
 
 		for ( Object value: list ){
 
