diff -rup Vuze_5720_source.orig/com/aelitis/azureus/util/ImportExportUtils.java Vuze_5720_source/com/aelitis/azureus/util/ImportExportUtils.java
--- Vuze_5720_source.orig/com/aelitis/azureus/util/ImportExportUtils.java	2016-05-02 18:16:02.000000000 +0100
+++ Vuze_5720_source/com/aelitis/azureus/util/ImportExportUtils.java	2016-05-23 02:49:34.859216438 +0100
@@ -371,7 +371,7 @@ public final class ImportExportUtils {
 		String		key,
 		String[]	data )
 	{
-		List	l = new JSONArray(data.length);
+		List	l = new JSONArray();
 		
 		map.put( key, l );
 		
diff -rup Vuze_5720_source.orig/com/aelitis/azureus/util/JSONUtils.java Vuze_5720_source/com/aelitis/azureus/util/JSONUtils.java
--- Vuze_5720_source.orig/com/aelitis/azureus/util/JSONUtils.java	2016-04-13 20:20:34.000000000 +0100
+++ Vuze_5720_source/com/aelitis/azureus/util/JSONUtils.java	2016-05-23 02:49:34.860216453 +0100
@@ -71,7 +71,7 @@ public class JSONUtils
 	 * @since 3.0.1.5
 	 */
 	public static JSONObject encodeToJSONObject(Map map) {
-		JSONObject newMap = new JSONObject((int)(map.size()*1.5));
+		JSONObject newMap = new JSONObject();
 
 		for (Map.Entry<String, Object> entry: ((Map<String,Object>)map).entrySet()){
 			String key 		= entry.getKey();
@@ -102,9 +102,7 @@ public class JSONUtils
 	 */
 	public static String encodeToJSON(Map map) {
 		JSONObject jobj = encodeToJSONObject(map);		
-		StringBuilder	sb = new StringBuilder(8192);
-		jobj.toString( sb );
-		return( sb.toString());
+		return( jobj.toString());
 	}
 
 	public static String encodeToJSON(Collection list) {
@@ -135,7 +133,7 @@ public class JSONUtils
 	 * @since 3.0.1.5
 	 */
 	private static JSONArray encodeToJSONArray(Collection list) {
-		JSONArray newList = new JSONArray(list.size());
+		JSONArray newList = new JSONArray();
 
 		for ( Object value: list ){
 
