--- org/gudy/azureus2/pluginsimpl/local/PluginInterfaceImpl.java.orig	2006-05-02 02:33:14.000000000 -0700
+++ org/gudy/azureus2/pluginsimpl/local/PluginInterfaceImpl.java	2006-05-12 09:58:07.000000000 -0700
@@ -48,7 +48,6 @@
 import org.gudy.azureus2.pluginsimpl.local.ui.config.ParameterRepository;
 import org.gudy.azureus2.pluginsimpl.local.ui.config.PluginConfigUIFactoryImpl;
 import org.gudy.azureus2.pluginsimpl.local.utils.*;
-import org.gudy.azureus2.pluginsimpl.local.update.*;
 import org.gudy.azureus2.plugins.ipc.IPCInterface;
 import org.gudy.azureus2.plugins.ipfilter.IPFilter;
 import org.gudy.azureus2.plugins.tracker.Tracker;
@@ -412,7 +411,7 @@
   public UpdateManager
   getUpdateManager()
   {
-  	return( UpdateManagerImpl.getSingleton( initialiser.getAzureusCore()));
+      return( null );
   }
 
   public PeerProtocolManager
