--- com/aelitis/azureus/util/win32/Win32Utils.java.orig	2007-02-14 11:57:30.000000000 -0800
+++ com/aelitis/azureus/util/win32/Win32Utils.java	2008-12-20 00:13:03.000000000 -0800
@@ -18,12 +18,12 @@
  *
  */
 
 package com.aelitis.azureus.util.win32;
 
-import org.gudy.azureus2.platform.win32.access.AEWin32Access;
-import org.gudy.azureus2.platform.win32.access.AEWin32Manager;
+//import org.gudy.azureus2.platform.win32.access.AEWin32Access;
+//import org.gudy.azureus2.platform.win32.access.AEWin32Manager;
 
 /**
  * @author TuxPaper
  * @created Feb 13, 2007
  *
@@ -33,17 +33,17 @@
 	/**
 	 * Get the location of Windows Media Player executable
 	 * @return
 	 */
 	public static String getWMP() {
-		AEWin32Access accessor = AEWin32Manager.getAccessor(true);
-		if (accessor == null) {
-			return null;
-		}
-		try {
-			return accessor.readStringValue(AEWin32Access.HKEY_LOCAL_MACHINE,
-					"SOFTWARE\\Microsoft\\Multimedia\\WMPlayer", "Player.Path");
-		} catch (Exception e) {
-		}
+//		AEWin32Access accessor = AEWin32Manager.getAccessor(true);
+//		if (accessor == null) {
+//			return null;
+//		}
+//		try {
+//			return accessor.readStringValue(AEWin32Access.HKEY_LOCAL_MACHINE,
+//					"SOFTWARE\\Microsoft\\Multimedia\\WMPlayer", "Player.Path");
+//		} catch (Exception e) {
+//		}
 		return null;
 	}
 }
