--- ./org/gudy/azureus2/core3/util/ShutdownHook.java~	2005-06-14 19:06:26.000000000 -0700
+++ ./org/gudy/azureus2/core3/util/ShutdownHook.java	2005-12-17 12:01:13.000000000 -0800
@@ -22,8 +22,8 @@
 
 package org.gudy.azureus2.core3.util;
 
-import sun.misc.Signal;
-import sun.misc.SignalHandler;
+// import sun.misc.Signal;
+// import sun.misc.SignalHandler;
 
 
 
@@ -40,44 +40,6 @@
    * @param handler
    */
   public static void install( final Handler handler) {
-    try{
-      final SignalHandler[] old_handlers = new SignalHandler[2];
-    
-      //ctrl-c signal
-      old_handlers[0] = Signal.handle( new Signal( "INT" ), new SignalHandler() {
-        public void handle( Signal sig ) {
-          try{
-            handler.shutdown( sig.getName() );  //main ShutdownHook.Handler
-            
-            if( old_handlers[0] != null && old_handlers[0] != SIG_DFL && old_handlers[0] != SIG_IGN ) {  //chain back to previous handler if one exists
-              old_handlers[0].handle( sig );
-            }
-          }
-          catch( Throwable t ) {
-            Debug.out( t );
-          }
-        }
-      });
-      
-      //os termination signal
-      old_handlers[1] = Signal.handle( new Signal( "TERM" ), new SignalHandler() {
-        public void handle( Signal sig ) {
-          try{
-            handler.shutdown( sig.getName() );  //main ShutdownHook.Handler
-            
-            if( old_handlers[1] != null && old_handlers[1] != SIG_DFL && old_handlers[1] != SIG_IGN ) {  //chain back to previous handler if one exists
-              old_handlers[1].handle( sig );
-            }
-          }
-          catch( Throwable t ) {
-            Debug.out( t );
-          }
-        }
-      });
-    }
-    catch( Throwable t ) {
-      Debug.out( t );
-    }
     
   }
   
