--- org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java.orig	2006-02-09 22:57:59.000000000 -0500
+++ org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java	2006-02-09 23:01:31.000000000 -0500
@@ -47,11 +47,8 @@
 import org.gudy.azureus2.plugins.*;
 import org.gudy.azureus2.pluginsimpl.local.launch.PluginLauncherImpl;
 import org.gudy.azureus2.pluginsimpl.local.ui.UIManagerImpl;
-import org.gudy.azureus2.pluginsimpl.local.update.*;
 import org.gudy.azureus2.pluginsimpl.local.utils.UtilitiesImpl;
 
-import org.gudy.azureus2.update.UpdaterUpdateChecker;
-
 
 
 /**
@@ -93,11 +90,6 @@
    					"azbpsharehoster", 
    					"ShareHoster",
 					"true" },
-   			{	 PluginManagerDefaults.PID_PLUGIN_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.pluginsimpl.update.PluginUpdatePlugin", 
-   					"azbppluginupdate", 
-   					"PluginUpdate",
-					"true" },
 			{	 PluginManagerDefaults.PID_CLIENT_ID, 
 				    "com.aelitis.azureus.plugins.clientid.ClientIDPlugin", 
 				    "azbpclientid", 
@@ -123,16 +115,6 @@
 					"azbpmagnet", 
 					"Magnet URI Handler",
 					"true" },
-			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.update.CoreUpdateChecker", 
-   					"azbpcoreupdater", 
-   					"CoreUpdater",
-					"true" },
-			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
-   					"org.gudy.azureus2.update.CorePatchChecker", 
-   					"azbpcorepatcher", 
-   					"CorePatcher",
-					"true" },
 	   		{	 PluginManagerDefaults.PID_JPC, 
 					"com.aelitis.azureus.plugins.jpc.JPCPlugin", 
 					"azjpc", 
@@ -297,11 +279,7 @@
   	
     listener 	= _listener;
     
-    UpdateManagerImpl.getSingleton( azureus_core );	// initialise the update manager
-       
     plugin_manager = PluginManagerImpl.getSingleton( this );
-    
-    UpdaterUpdateChecker.checkPlugin();
   }
   
   	public void 
@@ -823,16 +801,11 @@
 	      
 	      if ( load_failure != null ){
 	      		  
-	    	  	// don't complain about our internal one
-	    	  
-	    	  if ( !pid.equals(UpdaterUpdateChecker.getPluginID())){
-	    		  
 		      	String	msg = "Error loading plugin '" + pluginName + "' / '" + plugin_class_string + "'";
 		   	 
 		      	Logger.log(new LogAlert(LogAlert.UNREPEATABLE, msg, load_failure));
 	
 		      	System.out.println( msg + " : " + load_failure);
-		      }
 	      }
   		}
 	      
