diff -up azureus-4.5.0.0/org/gudy/azureus2/ui/swt/views/table/impl/TreeDelegate.java.no-Tree2 azureus-4.5.0.0/org/gudy/azureus2/ui/swt/views/table/impl/TreeDelegate.java
--- azureus-4.5.0.0/org/gudy/azureus2/ui/swt/views/table/impl/TreeDelegate.java.no-Tree2	2010-08-07 20:07:59.000000000 +0300
+++ azureus-4.5.0.0/org/gudy/azureus2/ui/swt/views/table/impl/TreeDelegate.java	2010-08-07 20:10:49.000000000 +0300
@@ -50,7 +50,7 @@ public class TreeDelegate implements Tab
 	Map<String, Object> data = new HashMap<String, Object>(5);
 
 	protected TreeDelegate(Composite parent, int style) {
-		this(Constants.isWindows ? new Tree2(parent, style) : new Tree(parent,
+		this(new Tree(parent,
 				style));
 	}
 
