--- com/aelitis/azureus/core/networkmanager/impl/udp/UDPConnectionSet.java.~1.12.~	2006-07-01 05:30:35.000000000 -0700
+++ com/aelitis/azureus/core/networkmanager/impl/udp/UDPConnectionSet.java	2006-07-08 11:13:04.000000000 -0700
@@ -333,7 +333,7 @@
     	
     	byte[]	temp = new byte[1024];
 	
-    	rc4_engine.processBytes( temp, temp );
+    	rc4_engine.processBytes( temp, 0, temp.length, temp, 0 );
     	
     	return( rc4_engine );
 	}
@@ -1683,7 +1683,7 @@
 	{
 		byte[]	bytes = intToBytes( i );
 		
-		cipher.processBytes( bytes, bytes );
+		cipher.processBytes( bytes, 0, bytes.length, bytes, 0 );
 		
 		return( bytesToInt( bytes, 0 ));
 	}
