--- org/gudy/azureus2/ui/swt/mainwindow/MainMenu.java.orig	2006-03-28 04:35:35.000000000 -0800
+++ org/gudy/azureus2/ui/swt/mainwindow/MainMenu.java	2006-03-28 04:41:30.000000000 -0800
@@ -347,26 +347,6 @@
 			menu_plugin_logViews.setMenu(pluginLogsMenu);
 			menu_plugin_logViews.setData("EOL", "1");
 
-      new MenuItem(pluginMenu, SWT.SEPARATOR);
-      
-      MenuItem plugins_install_wizard = new MenuItem(pluginMenu, SWT.NULL);
-      KeyBindings.setAccelerator(plugins_install_wizard, "MainWindow.menu.plugins.installPlugins");
-      Messages.setLanguageText(plugins_install_wizard, "MainWindow.menu.plugins.installPlugins"); //$NON-NLS-1$
-      plugins_install_wizard.addListener(SWT.Selection, new Listener() {
-        public void handleEvent(Event e) {
-          new InstallPluginWizard(mainWindow.getAzureusCore(), display);
-        }
-      });
-      
-      MenuItem plugins_uninstall_wizard = new MenuItem(pluginMenu, SWT.NULL);
-      KeyBindings.setAccelerator(plugins_uninstall_wizard, "MainWindow.menu.plugins.uninstallPlugins");
-      Messages.setLanguageText(plugins_uninstall_wizard, "MainWindow.menu.plugins.uninstallPlugins"); //$NON-NLS-1$
-      plugins_uninstall_wizard.addListener(SWT.Selection, new Listener() {
-        public void handleEvent(Event e) {
-          new UnInstallPluginWizard(mainWindow.getAzureusCore(), display);
-        }
-      });
-
       // standard items
       if(Constants.isOSX) {
           // Window menu
@@ -452,18 +432,6 @@
       
       new MenuItem(helpMenu,SWT.SEPARATOR);
       
-      if ( !SystemProperties.isJavaWebStartInstance()){
-        MenuItem help_checkupdate = new MenuItem(helpMenu, SWT.NULL);
-        KeyBindings.setAccelerator(help_checkupdate, "MainWindow.menu.help.checkupdate");
-        Messages.setLanguageText(help_checkupdate, "MainWindow.menu.help.checkupdate"); //$NON-NLS-1$
-        help_checkupdate.addListener(SWT.Selection, new Listener() {
-        	public void handleEvent(Event e) {
-        		mainWindow.getShell().setFocus();
-        		UpdateMonitor.getSingleton( mainWindow.getAzureusCore()).performCheck(true);
-        	}
-        });
-      }
-
       MenuItem help_donate = new MenuItem(helpMenu, SWT.NULL);
       Messages.setLanguageText(help_donate, "MainWindow.menu.help.donate"); //$NON-NLS-1$
       help_donate.addListener(SWT.Selection, new Listener() {
