--- org/gudy/azureus2/ui/swt/views/configsections/ConfigSectionInterfaceDisplay.java.orig	2006-02-15 05:36:31.000000000 -0800
+++ org/gudy/azureus2/ui/swt/views/configsections/ConfigSectionInterfaceDisplay.java	2006-05-26 09:55:19.000000000 -0700
@@ -79,7 +79,7 @@
 
 		BooleanParameter bpCustomTab = new BooleanParameter(cLook, "useCustomTab",
 				true, MSG_PREFIX + "useCustomTabs");
-		Control cFancyTab = new BooleanParameter(cLook, "GUI_SWT_bFancyTab", true,
+		Control cFancyTab = new BooleanParameter(cLook, "GUI_SWT_bFancyTab", false,
 				MSG_PREFIX + "useFancyTabs").getControl();
 
 		Control[] controls = { cFancyTab };
--- org/gudy/azureus2/core3/config/impl/ConfigurationDefaults.java.~1.196.~	2006-05-26 09:16:13.000000000 -0700
+++ org/gudy/azureus2/core3/config/impl/ConfigurationDefaults.java	2006-05-26 09:56:44.000000000 -0700
@@ -174,7 +174,7 @@
     def.put("Use default data dir", FALSE);	
     def.put("Default save path", "" );
     
-    def.put("GUI_SWT_bFancyTab", TRUE);
+    def.put("GUI_SWT_bFancyTab", FALSE);
     def.put("GUI_SWT_bAlternateTablePainting", FALSE);
     def.put("update.start",TRUE);
     def.put("update.periodic",TRUE);
