diff -up azureus-4.5.0.0/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java.no-updates-PluginInitializer azureus-4.5.0.0/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java
--- azureus-4.5.0.0/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java.no-updates-PluginInitializer	2010-07-31 01:00:40.000000000 +0300
+++ azureus-4.5.0.0/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java	2010-08-05 20:33:30.000000000 +0300
@@ -40,12 +40,8 @@ import org.gudy.azureus2.core3.util.*;
 import org.gudy.azureus2.plugins.*;
 import org.gudy.azureus2.pluginsimpl.local.launch.PluginLauncherImpl;
 import org.gudy.azureus2.pluginsimpl.local.ui.UIManagerImpl;
-import org.gudy.azureus2.pluginsimpl.local.update.UpdateManagerImpl;
 import org.gudy.azureus2.pluginsimpl.local.utils.UtilitiesImpl;
 import org.gudy.azureus2.pluginsimpl.local.utils.UtilitiesImpl.runnableWithException;
-import org.gudy.azureus2.update.UpdaterUpdateChecker;
-import org.gudy.azureus2.update.UpdaterUtils;
-
 
 import com.aelitis.azureus.core.*;
 import com.aelitis.azureus.core.versioncheck.VersionCheckClient;
@@ -97,12 +93,12 @@ PluginInitializer
    					"ShareHoster",
 					"true",
 					"false"},
-   			{	 PluginManagerDefaults.PID_PLUGIN_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.pluginsimpl.update.PluginUpdatePlugin", 
-   					"azbppluginupdate", 
-   					"PluginUpdate",
-					"true",
-					"true"},
+//    			{	 PluginManagerDefaults.PID_PLUGIN_UPDATE_CHECKER, 
+//    					"org.gudy.azureus2.pluginsimpl.update.PluginUpdatePlugin", 
+//    					"azbppluginupdate", 
+//    					"PluginUpdate",
+// 					"true",
+// 					"true"},
 			{	 PluginManagerDefaults.PID_UPNP, 
 				    "com.aelitis.azureus.plugins.upnp.UPnPPlugin", 
 				    "azbpupnp", 
@@ -127,18 +123,18 @@ PluginInitializer
 					"Magnet URI Handler",
 					"true",
 					"false"},
-			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.update.CoreUpdateChecker", 
-   					"azbpcoreupdater", 
-   					"CoreUpdater",
-					"true",
-					"true"},
-			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
-   					"org.gudy.azureus2.update.CorePatchChecker", 
-   					"azbpcorepatcher", 
-   					"CorePatcher",
-					"true",
-					"true"},
+// 			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
+//    					"org.gudy.azureus2.update.CoreUpdateChecker", 
+//    					"azbpcoreupdater", 
+//    					"CoreUpdater",
+// 					"true",
+// 					"true"},
+// 			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
+//    					"org.gudy.azureus2.update.CorePatchChecker", 
+//    					"azbpcorepatcher", 
+//    					"CorePatcher",
+// 					"true",
+// 					"true"},
 	   		{	 PluginManagerDefaults.PID_PLATFORM_CHECKER, 
    					"org.gudy.azureus2.platform.PlatformManagerPluginDelegate", 
    					"azplatform2", 
@@ -457,7 +453,7 @@ PluginInitializer
   	
   	core_operation 	= _core_operation;
     
-    UpdateManagerImpl.getSingleton( azureus_core );	// initialise the update manager
+//     UpdateManagerImpl.getSingleton( azureus_core );	// initialise the update manager
        
     plugin_manager = PluginManagerImpl.getSingleton( this );
     
@@ -479,7 +475,7 @@ PluginInitializer
     	}
     }
     
-    UpdaterUtils.checkBootstrapPlugins();
+//     UpdaterUtils.checkBootstrapPlugins();
   }
   
   protected void
@@ -1233,9 +1229,6 @@ PluginInitializer
 		      if ( load_failure != null ){
 		    	  plugin_interface.setAsFailed();
 		    	  
-		    	// don't complain about our internal one
-		    	  if ( !pid.equals(UpdaterUpdateChecker.getPluginID())){
-		    		  
 			      	String msg = "Error loading plugin '" + pluginName + "' / '" + plugin_class_string + "'";
 			      	LogAlert la;
 			      	if (load_failure instanceof UnsupportedClassVersionError) {
@@ -1247,7 +1240,6 @@ PluginInitializer
 			      	Logger.log(la);
 		
 			      	System.out.println( msg + ": " + load_failure);
-			      }
 		      }
 	  		}
   		}
