--- org/gudy/azureus2/core3/security/impl/SESecurityManagerImpl.java~	2005-11-03 07:32:04.000000000 -0800
+++ org/gudy/azureus2/core3/security/impl/SESecurityManagerImpl.java	2005-12-17 19:33:27.000000000 -0800
@@ -86,7 +86,7 @@
 		
 		try{
 			Security.addProvider((java.security.Provider)
-				Class.forName("com.sun.net.ssl.internal.ssl.Provider").newInstance());
+				Class.forName("org.metastatic.jessie.provider.Jessie").newInstance());
 			
 		}catch( Throwable e ){
 			
