--- com/aelitis/azureus/core/update/impl/AzureusRestarterImpl.java.orig	2008-12-20 00:45:25.000000000 -0800
+++ com/aelitis/azureus/core/update/impl/AzureusRestarterImpl.java	2008-12-20 00:45:42.000000000 -0800
@@ -217,10 +217,11 @@
 	private boolean
 	win32NativeRestart(
 		PrintWriter	log,
 		String		exec )
 	{
+          return true;
 	}
 	
 
 	private String getExeUpdater(PrintWriter log) {
 		try {
